src/queryinterface.js

```diff
@@ -106,8 +121,14 @@ QueryInterface.prototype.$queueError = QueryInterface.prototype.$queueFailure;
  **/
 QueryInterface.prototype.$clearQueue = function (options) {
 	options = options || {};
-	this._results = [];
-	
+
+	if (options.clearResults !== false) {
```

Comment:
The defaults are `clearResults:true` and `clearHandlers:true`.

Other options would be to use `skipResults` and `skipHandlers`, or create a separate method to clear the handlers queue.